### PR TITLE
SWAG-20562 comments to eslint rules

### PR DESCRIPTION
**Pull request by change-my-username**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -23,18 +23,20 @@ module.exports = {
...
+    'filenames-simple/naming-convention': ['error', { rule: 'kebab-case' }], // +
+    'prettier/prettier': 'error', // REDUNDANT, it is already error in eslint-plugin-prettier
+    'operator-linebreak': ['error', 'after'], // +
+    '@typescript-eslint/no-shadow': ['error'], // +, also, need to set "no-shadow": "off" according to docs
     '@typescript-eslint/no-unused-vars': [
+      // +
       'error',
       {
-        args: 'none',
+        args: 'none', // probably, should be "after-used"
```
Comment (cyxn) [inline index.js:34]: dismissing that in favor of having functions accepting object arguments

File: index.js [hunk 2]
```diff
@@ -70,79 +74,83 @@ module.exports = {
         },
       },
     ],
-    '@typescript-eslint/member-ordering': 'off',
-    '@typescript-eslint/naming-convention': 'off',
-    '@typescript-eslint/no-empty-function': 'error',
-    '@typescript-eslint/no-empty-interface': 'off',
+    '@typescript-eslint/member-ordering': 'off', // order interface and class fields, could be useful
```
Comment (cyxn) [inline index.js:77]: dismissing that

File: index.js [hunk 3]
```diff
@@ -70,79 +74,83 @@ module.exports = {
...
+    '@typescript-eslint/naming-convention': 'off', // probably, useful, but hard to implement
```
Comment (cyxn) [inline index.js:78]: probably later or never

File: index.js [hunk 4]
```diff
@@ -70,79 +74,83 @@ module.exports = {
...
+    '@typescript-eslint/no-empty-function': 'error', // +
+    '@typescript-eslint/no-empty-interface': 'off', // probably, should be "error", DEPRECATED in favor of @typescript-eslint/no-empty-object-type
```
Comment (cyxn) [inline index.js:80]: probably later or never

File: index.js [hunk 5]
```diff
@@ -70,79 +74,83 @@ module.exports = {
...
     '@typescript-eslint/no-unused-expressions': [
+      // +
       'error',
       {
         allowShortCircuit: true,
       },
     ],
-    '@typescript-eslint/no-var-requires': 'off',
-    '@typescript-eslint/prefer-namespace-keyword': 'error',
+    '@typescript-eslint/no-var-requires': 'off', // +, DEPRECATED in favor of @typescript-eslint/no-require-imports
```
Comment (cyxn) [inline index.js:96]: probably later move to `@typescript-eslint/no-require-imports`

File: index.js [hunk 6]
```diff
@@ -70,79 +74,83 @@ module.exports = {
...
-    'security/detect-object-injection': 'off',
-    'spellcheck/spell-checker': mergeSpellCheckDictionary(),
+    '@typescript-eslint/semi': ['off', null], // should be removed because of it is managed by eslint-plugin-prettier
+    '@typescript-eslint/type-annotation-spacing': 'error', // should be removed because of it is managed by eslint-plugin-prettier
+    'arrow-parens': ['off', 'always'], // should be removed because of it is managed by eslint-plugin-prettier
+    'brace-style': ['error', '1tbs'], // +
+    'comma-dangle': 'off', // should be removed because of it is managed by eslint-plugin-prettier
+    'default-case': 'error', // +
+    'eol-last': 'error', // +
+    eqeqeq: ['off', 'smart'], // "off" option disables this rule, so "smart" is pointless, probably, it should be ['error', 'smart']
```
Comment (cyxn) [inline index.js:113]: for later

File: index.js [hunk 7]
```diff
@@ -70,79 +74,83 @@ module.exports = {
...
+    'id-blacklist': 'off', // +, DEPRECATED in favor of id-denylist
```
Comment (cyxn) [inline index.js:114]: for later to move to another rule

File: index.js [hunk 8]
```diff
@@ -70,79 +74,83 @@ module.exports = {
...
+    'id-match': 'off', // +, match identifier names to some reg exp. Not needed. Similar to @typescript-eslint/naming-convention
+    'linebreak-style': 'off', // should be removed because of it is managed by eslint-plugin-prettier
+    'max-len': 'off', // should be removed because of it is managed by eslint-plugin-prettier
+    'new-parens': 'off', // should be removed because of it is managed by eslint-plugin-prettier
+    'newline-per-chained-call': 'off', // +, not managed by eslint-plugin-prettier
+    'no-caller': 'error', // +
+    '@typescript-eslint/ban-types': 'warn', // OUTDATED, https://typescript-eslint.io/rules/ban-types/
```
Comment (cyxn) [inline index.js:121]: for later to move to another rule

File: index.js [hunk 9]
```diff
@@ -70,79 +74,83 @@ module.exports = {
...
+    'no-console': ['error'], // +
+    'no-empty': 'error', // +
+    'no-eval': 'error', // +
+    'no-extra-semi': 'off', // should be removed because of it is managed by eslint-plugin-prettier
+    'no-fallthrough': 'error', // +
+    'no-irregular-whitespace': 'off', // should be removed because of it is managed by eslint-plugin-prettier
+    'no-multiple-empty-lines': 'error', // should be removed because of it is managed by eslint-plugin-prettier
+    'no-new-wrappers': 'error', // +
+    'no-null/no-null': 'off', // +, if it is "off", maybe eslint-plugin-no-null should be completly removed ?
```
Comment (cyxn) [inline index.js:130]: yes, let's remove plugin

File: index.js [hunk 10]
```diff
@@ -70,79 +74,83 @@ module.exports = {
...
+    'no-redeclare': 'error', // +
+    'no-trailing-spaces': 'error', // should be removed because of it is managed by eslint-plugin-prettier
+    'no-undef-init': 'off', // +
+    'no-underscore-dangle': 'off', // probably, should be error to have consistent code base, but debatable (at least 3 errors found)
```
Comment (cyxn) [inline index.js:134]: for later

File: index.js [hunk 11]
```diff
@@ -70,79 +74,83 @@ module.exports = {
...
+    'prefer-object-spread': 'error', // +
+    curly: 'error', // should be removed because of it is managed by eslint-plugin-prettier
+    'quote-props': 'off', // DEPRECATED, managed by eslint-plugin-prettier
+    'react/jsx-curly-spacing': 'off', // +
+    'react/jsx-equals-spacing': 'off', // +
+    'react/jsx-wrap-multilines': 'off', // +
+    'space-before-function-paren': 'off', // +
+    'no-only-tests/no-only-tests': 'error', // +
+    'space-in-parens': ['off', 'never'], // should be removed because of it is managed by eslint-plugin-prettier
+    'security/detect-object-injection': 'off', // probably should be error, but there are a lot of errors (at least 2000 errors found)
```
Comment (cyxn) [inline index.js:145]: let's turn it off

**Review by cyxn — CHANGES_REQUESTED** [review]

File: index.js [hunk 12]
```diff
@@ -50,107 +53,66 @@ module.exports = {
         ],
       },
     ],
-    '@typescript-eslint/ban-ts-ignore': ['off'], // should be removed as it was replaced with ban-ts-comment (from 3.0.0)
-    '@typescript-eslint/ban-ts-comment': ['off'], // probably, should be warn
-    '@typescript-eslint/dot-notation': ['error'], // +, also, need to set "dot-notation": "off" according to docs
+    '@typescript-eslint/no-explicit-any': 'warn',
```
Comment (cyxn) [inline index.js:56]: I think it was `error`, no?
I believe we put it to `warn` in core swag app only

File: index.js [hunk 13]
```diff
@@ -50,107 +53,66 @@ module.exports = {
...
+    '@typescript-eslint/ban-ts-comment': ['warn'],
```
Comment (cyxn) [inline index.js:57]: same as above

**Review by cyxn — CHANGES_REQUESTED** [review]

File: index.js (hunk 12 above)
Comment (change-my-username) [reply index.js:56]: I set it to warn because a lot of errors across all repos

File: index.js (hunk 13 above)
Comment (change-my-username) [reply index.js:57]: same as above

File: index.js
[the remark this answers is not in the document]
Comment (change-my-username) [reply index.js:1]: `no-explicit-any` is set to error in `plugin:@typescript-eslint/recommended`
`explicit-module-boundary-types` is absent in `plugin:@typescript-eslint/recommended` and also produces a lot of errors across all repos as `no-explicit-any` if it set to error

[the remark this answers is not in the document]
Comment (change-my-username) [reply index.js:1]: Probably, These rules could be commented as `// need to fix `

**Review by cyxn — APPROVED** [review]

**Pull request merged**